xinference/core/supervisor.py

```diff
@@ -870,6 +870,9 @@ async def _check_dead_nodes(self):
                             address,
                             dead_models,
                         )
+                        for model_uid in dead_models:
+                            self._model_uid_to_replica_info.pop(model_uid[:-4], None)
+                            self._replica_model_uid_to_worker.pop(model_uid, None)
```

Comment:
Please also consider to remove info in the function `remove_worker`. 
`remove_worker` is the logic of the supervisor when the worker is normally exited by ctrl-c. From there it is possible to remove worker related information directly. 
On the one hand, it forms a double insurance policy with the worker detection logic. 
On the other hand, some users may not turn on worker detection.